File: B/Python/143.py
N, M = map(int, input().split())

winners = [[] for _ in range(N+1)]

for i in range(M):
    x, y = map(int,input().split())
    winners[x].append(y)
    # 勝った場合はこれまでの人を追加
    if len(winners[y]) != 0:
        for j in winners[y]:
            winners[x].append(j)
    # 負けた場合は初期化
    winners[y] = []

# 勝った数をそれぞれ追加
counts_list = []
for i in range(N):
    counts_list.append(len(winners[i+1]))

# 最大勝利数の算出
max_winner = max(counts_list)
for i in range(N):
    if max_winner == counts_list[i]:
        print(i+1)
        

File: B/JavaScript/143.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split("\n");

const [N, M] = input[0].split(" ").map(Number);
let winners = Array.from({ length: N+1}, () => []);

for (let i = 0; i < M; i++){
    let [x, y] = input[i+1].split(" ").map(Number);

    winners[x].push(y);
    if (winners[y].length > 0){
        winners[x].push(...winners[y]);
    }
    winners[y] = [];
}

let counts = Array(N+1).fill(0);
for (let i = 1; i <= N; i++){
    counts[i] = winners[i].length;
}

let maxwinner = Math.max(...counts.slice(1));

for (let i = 1; i <= N; i++){
    if (counts[i] === maxwinner){
        console.log(i);
    }
}